### DISCUSSION TITLE
Require PikePDF >= 6

### DISCUSSION THREAD

**Discussion by jeromerobert** [pdfarranger/pdfarranger#1013] [General]
I plan to drop PikePDF < 6 support for PDF-Arranger 1.11. Is anyone concerned about it ?

**Comment by dreua** [comment]
Nope, distros with old pikepdf can use old PDF Arranger. We've been quite good regarding bugs if the decreasing number of reports is any measure for that, so the last versions will be quite usable for anyone with older pikepdf. The impatient for new features will need to switch/upgrade distros or use the flatpak, that is just how it works.

**Comment by dreua** [comment]
I stumbled upon the idea to use versioning differently for end used applications in a way also that increases the major version when requirements change. It sounds like an interesting and potentially useful idea to me. The semver doesn't really make sense for a GUI application unless you have really huge changes in which case you can still bump major anyway.
The article explains it has more arguments and detail: https://uglow.medium.com/making-sense-of-semantic-versioning-for-end-user-software-applications-a3049d97478b

I think we could adopt this and make it PDF Arranger 2 but I'm fine either way.

**Reply by dreua** [reply]
Either way we need to remember to mention it in the release.

**Comment by dreua** [comment]
One more thought: Shall we also declare a minimum Python version? I'd like that, also as a guideline when changing code. One example would be a feature of the `@dataclass` decorator added in Python 3.7 (see #917)
